presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -6408,4 +6410,40 @@ public void testPrefilterGroupByLimitNotFiredForLowCardinalityKeys()
         plan = computeActual(prefilter, "explain(type distributed) select count(custkey), orderkey from orders group by orderkey limit 100000");
         assertTrue(((String) plan.getOnlyValue()).toUpperCase().indexOf("MAP_AGG") == -1);
     }
+
+    @Test
+    public void testConversionOfCrossJoinWithLeftOuterJoinToInnerJoin()
+    {
+        Session useDefaultLiteralCoalesce = getSession();
+        Session useCrossJoinCoalesce = Session.builder(getSession())
+                .setSystemProperty(USE_DEFAULTS_FOR_CORRELATED_AGGREGATION_PUSHDOWN_THROUGH_OUTER_JOINS, Boolean.toString(false))
+                .build();
+        ImmutableList.Builder<String> queries = ImmutableList.builder();
+        ImmutableList.Builder<String> expectedResults = ImmutableList.builder();
+
+        queries.add("select count(*) from nation n where (select count(*) from customer c where n.nationkey=c.nationkey)>0");
```

Comment:
I would add comments about what each case tests, eg. max and sum use default null,  approx_distinct, max_by won't use defaults  will do the cross join behavior, count(*) doesn't trigger the optimization.  maybe add a test for count(col) too.

Reply:
Done. 
Fyi - But in this case the optimization will apply to count(*) since it gets rewritten to count(non_null). It will only not kick in when you try to push a count(*) directly through the join - but as you point out earlier, there is another check that prevents this from happening